Update README using the one from the enarx.github.io repo (#154)
enarx/enarx: README.md

# Enarx

## Introduction
Enarx is an application deployment system enabling applications to run within Trusted Execution Environments (TEEs) without rewriting for particular platforms or SDKs. It handles attestation and delivery into a run-time “Keep” based on WebAssembly, offering developers a wide range of language choices for implementation. Enarx is CPU-architecture independent, enabling the same application code to be deployed across multiple targets, abstracting issues such as cross-compilation and differing attestation mechanisms between hardware vendors. Work is currently underway on AMD SEV and Intel SGX.

> We've known for a long time that we need encryption for data at rest and in transit: Enarx helps you do encryption for data in use.

Reviewer: ```suggestion
We've known for a long time that we need encryption for data at rest and in transit: Enarx helps you do encryption for data as well as algorithms in use.
```

Author: Good point. Will incorporate.

> We've known for a long time that we need encryption for data at rest and in transit: Enarx helps you do encryption for data in use.

Reviewer 2: I'd suggest being a little less colloquial here. Perhaps:

```suggestion
Enarx aims to transparently enable encryption for code and data:
  * in transit
  * at rest
  * in use
```

Author: I don't think i agree here: i think we impart less context and don't make our intentions with Enarx as clear by just saying it aims to transparently enable encryption for code and data. Arguably, LUKS already does just that. Sure, it's not for code or data in use.
Which is why i think the context of encrypting data at rest, data in transit and data in use is helpful.

Adding "code and data" is useful however, and i've incorporated Stefan's suggestion.

> [Read more](https://github.com/enarx/enarx.github.io/wiki/Enarx-Introduction)

Author: thanks, well spotted, will amend

## Project aim
Create a way to create and run "private, fungible, serverless" applications using Trusted Execution Environments (TEEs). In other words, to provide a platform abstraction for TEEs.

They should:

 * be confidential from the host and other workloads;
 * be able to run on
   * multiple silicon architectures; 
   * multiple server types;
   * public cloud or private cloud;

>  * employ serverless deployment and execution mechanisms.

Reviewer: Rather than the buzzword I'd describe the properties this implies in your understanding.

Author: This is a good point @steveeJ, however i think that will require to put thought into it than what this issue and PR intended to do. I'll resolve for now but open an issue to track this.

The mechanism itself should be capable of being FIPS-certified.

## Getting Started

> For more information, please try our [wiki](https://github.com/enarx/enarx/wiki), which includes further details on how to get more information.  It's also where we plan to keep up-to-date project information.

Author: The logic so far is have people land on the README, give them a bit of information, then refer them to the full documentation for more.

I agree there are more decisions to be made regarding the wiki's home vs. the wiki's introduction page. But this PR intended to just resolve the initial issue of having an outdated README on enarx/enarx.

Does that resolve your concerns?

> For more information, please try our [wiki](https://github.com/enarx/enarx/wiki), which includes further details on how to get more information.  It's also where we plan to keep up-to-date project information.

Reviewer 3: Just something I noted. It's not a blocker in any case.

Building the various components of Enarx is currently complex: we are working on this.  Please contact us for help.

(2020-01-01) It is also worth mentioning that it isn't yet possible to run Enarx.  We're working hard on it and we'd love people to work with us. We hope to be adding more information very soon, to allow you to get started.

## Authors

* **Mike Bursell** - *Initial work* - [MikeCamel](https://github.com/MikeCamel)

> See also the list of [people](https://github.com/orgs/enarx/people) who participated in this project.

Reviewer: I'd use present tense here. The participation is ongoing :-)

Reviewer 4: Indeed.  please fix.

Author: Fixed in latest pushed branch.
